### ISSUE TITLE
[BUG] STRTCPSVR fails, though `sc` command does not

### ISSUE THREAD

**Issue by thebeardedgeek** [ThePrez/ServiceCommander-IBMi#35]
**Describe the bug**
```STRTCPSVR SERVER(*SC) INSTANCE('node-red')``` returns two errors ```ERROR: Timed out waiting for service 'NodeRED server' to start``` and ```node: A file or directory in the path name does not exist.```. I can run ```sc start node-red``` from SSH and the process starts normally.

**To Reproduce**
I am trying to start node-red using ```STRTCPSVR SERVER(*SC) INSTANCE('node-red')```. It kicks off some processes in WRKACTJOB but the terminal session shows this ```ERROR: Timed out waiting for service 'NodeRED server' to start``` and the job ends. When I check QPRINT I found this error ```node: A file or directory in the path name does not exist.```

**Expected behavior**
I would like ```STRTCPSVR SERVER(*SC) INSTANCE('node-red')``` to start the node-red task.

**Verbose output**
There is no verbose output but I thought this might help:
```
LC_ALL=EN_US.UTF-8                                                            
QIBM_MULTI_THREADED=Y                                                         
SBMJOB_JOBNAME=NODERED                                                        
LANG=EN_US                                                                    
TZ=<CST>6<CDT>,M3.2.0,M11.1.0                                                 
OLDPWD=/home/MATTDEV
PASE_PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
PASE_USRGRP_LIMITED=N                                                         
SBMJOB_OPTS=JOB(NODERED) USER(MATTDEV) JOBQ(QSYS/QUSRNOMAX)                
QIBM_USE_DESCRIPTOR_STDIO=N                                                   
PASE_LOCPATH=/usr/lib/nls/loc                                                 
PWD=/home/MATTDEV
HOME=/home/MATTDEV
PASE_LANG=EN_US                                                               
PASE_LC__FASTMSG=true                                                         
PASE_NLSPATH=/usr/lib/nls/msg/%L/%N:/usr/lib/nls/msg/%L/%N.cat:/usr/lib/nls/ms
QIBM_PASE_CCSID=1208                                                          
USRGRP_LIMITED=N                                                              
LOCPATH=/usr/lib/nls/loc                                                      
SHLVL=1                                                                       
LC__FASTMSG=true                                                              
LOGNAME=MATTDEV
QIBM_IFS_OPEN_MAX=66000                                                       
PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
IBM_VE2_LOCATION=/QIBM/ProdData/VE2                                           
PASE_TZ=<CST>6<CDT>,M3.2.0,M11.1.0                                            
_=/QOpenSys/usr/bin/env                                                       
NLSPATH=/usr/lib/nls/msg/%L/%N:/usr/lib/nls/msg/%L/%N.cat:/usr/lib/nls/msg/EN_US/%N:/usr/lib/nls/msg/EN_US/%N.cat
```

**Additional context**
I am using dotfiles but that is just for my user. I have populated the .profile and .bash_profile files for my user; which is also the user executing these processes.

**Comment by ThePrez** [comment]
This is because bash did not run your `.bash_profile` in this context, and the [ibmi-dotfiles project](https://github.com/jbh/ibmi-dotfiles) uses `.bash_profile` instead of `.profile`. 

Refer to ye olde flowchart 
http://www.solipsys.co.uk/new/BashInitialisationFiles.html

The fix should be relatively simple: `sc` should use `-l`/`--login` on the bash invocation when running in batch to match the behavior of the interactive bash environment.

**Comment by ThePrez** [comment]
I hope to get to this later next week, and will include the fix in the next release

**Comment by ThePrez** [comment]
@thebeardedgeek , can you please pull the latest code of the `strtcpsvr` branch and re-test. I believe this will be resolved

**Comment by thebeardedgeek** [comment]
@ThePrez it is still not able to find the path.
error: `node: A file or directory in the path name does not exist.`

I am still testing this, but it looks like node is being called but it is struggling to find a file or directory. Any way to get more info? Am I understanding this correctly?

**Comment by ThePrez** [comment]
Yeah, the problem is that the PATH isn't set up properly. I thought my latest changes (launching `bash -l`) would fix that, but no dice. I think I need to set up a recreate using ibmi-dotfiles

**Comment by ThePrez** [comment]
Just to make sure it's cloned propetly, can you do a `git log` and make sure your most recent commit is 519264e9814063e02fbbe4735415a4a6dad10f84 ?

**Comment by ThePrez** [comment]
Also, just to cover all the bases, can you give me the output of `sc info node-red`?

**Comment by ThePrez** [comment]
@thebeardedgeek , one more request.... Please get verbose output by setting the SC_OPTIONS env var
`ADDENVVAR ENVVAR(SC_OPTIONS) VALUE('-v')`
That will trigger a trace point that will help me isolate the problem.

**Comment by thebeardedgeek** [comment]
I verified I am on the correct commit version and here is the other info you asked for. It doesn't appear that node is in my path based on the verbose log but I'll let you be the judge of that.

![image](https://user-images.githubusercontent.com/9466407/114206039-89830380-9920-11eb-9f74-f2e478b433a1.png)

![image](https://user-images.githubusercontent.com/9466407/114206103-9d2e6a00-9920-11eb-9633-aefebaa186d4.png)
[QPRINT_MSEEBERGER_NODERED_364453_1.txt](https://github.com/ThePrez/ServiceCommander-IBMi/files/6286981/QPRINT_MSEEBERGER_NODERED_364453_1.txt)
[QPRINT_MSEEBERGER_NODERED_364454_1.txt](https://github.com/ThePrez/ServiceCommander-IBMi/files/6286982/QPRINT_MSEEBERGER_NODERED_364454_1.txt)
[QPRINT_MSEEBERGER_QPADEV000C_364446_1.txt](https://github.com/ThePrez/ServiceCommander-IBMi/files/6286983/QPRINT_MSEEBERGER_QPADEV000C_364446_1.txt)

**Comment by ThePrez** [comment]
Yep, your PATH in the trace is indeed missing `/QOpenSys/pkgs/bin`. That means that the proper PATH was lost before the Java code tried submitting the command to batch. So, it was lost (or unset) somewhere in the STRTCPSVR -> bash -> java flow. 🤔 

I'm suprised the error isn't `-bash: node-red: command not found`. You must have a `/QOpenSys/usr/bin/node-red` but not a `/QOpenSys/usr/bin/node`? Is that true? 

An aside, I suspect running `nodever` will work around your problem, but please don't run it until we debug this issue!

**Comment by ThePrez** [comment]
More debug:
Please run `CALL PGM(QP2SHELL2) PARM('/QOpenSys/pkgs/bin/bash' '-l' '-c' 'env> env.txt')` and give us the resulting `env.txt` (home dir)

**Comment by thebeardedgeek** [comment]
Here is the file
[env.txt](https://github.com/ThePrez/ServiceCommander-IBMi/files/6289867/env.txt)

**Comment by ThePrez** [comment]
Thanks! That does confirm that the path is getting "lost" (or more accurately, not set) before the sc code even gets involved. Scratching chin...

**Comment by ThePrez** [comment]
I think the problem actually stems from here: https://github.com/jbh/ibmi-dotfiles/blob/22f8005f2cbc07a18d9e82fc7c68c05dff8ffd0f/system/bash_profile.symlink#L6

It checks if bash is running by checking the `BASH_VERSION` env var, which is not set for you. @jbh, am I chasing the right train of thought?

**Comment by ThePrez** [comment]
Next experiment: run `CALL PGM(QP2SHELL2) PARM('/QOpenSys/pkgs/bin/bash' '-l' '-i' '-c' 'env> env2.txt')` and give us the env2.txt

**Comment by thebeardedgeek** [comment]
Looks like the path is there now. Here is the file:
[env2.txt](https://github.com/ThePrez/ServiceCommander-IBMi/files/6296545/env2.txt)

**Comment by ThePrez** [comment]
@thebeardedgeek, can you please edit your `$HOME/.profile` as done by the patch [here](https://github.com/jbh/ibmi-dotfiles/pull/16/files)?

Then, please run CALL PGM(QP2SHELL2) PARM('/QOpenSys/pkgs/bin/bash' '-l' '-c' 'env> env3.txt') and give us the resulting env3.txt (home dir)

**Comment by thebeardedgeek** [comment]
I updated the .bash_profile with the recommended change
![image](https://user-images.githubusercontent.com/9466407/114875656-1fa4a700-9dc3-11eb-89f3-854cabc3054a.png)

Here is env3.txt. It doesn't have node in the path
[env3.txt](https://github.com/ThePrez/ServiceCommander-IBMi/files/6318287/env3.txt)

**Comment by ThePrez** [comment]
Sorry, @thebeardedgeek . My test environment was contaminated by a `/QOpenSys/etc/profile`

I'm now sure the problem is here:
https://github.com/jbh/ibmi-dotfiles/blob/22f8005f2cbc07a18d9e82fc7c68c05dff8ffd0f/system/bashrc.symlink#L25

`ibmi-dotfiles` explicitly only sets the PATH if it's an interactive shell.

**Comment by thebeardedgeek** [comment]
@ThePrez I commented out `#case $- in *i* )` and `#esac` and it started successfully using `STRTCPSVR SERVER(*SC) INSTANCE('node-red')`
![image](https://user-images.githubusercontent.com/9466407/114928881-bb9cd580-9df8-11eb-9f9a-bff6a2d27ad0.png)

**Comment by ThePrez** [comment]
Resolved by fixes #33 and the ibmi-dotfiles fix. Please reopen if the problem persists. Thanks for your patience on this one!
